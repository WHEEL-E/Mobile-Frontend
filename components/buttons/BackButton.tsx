import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";
import { InputBackButtonProps } from "../../utilities/buttonsUtils";

export const BackButton = (props: InputBackButtonProps) => {
  const { onPress } = props;

  return (
    <TouchableOpacity style={styles.button}>
      <Ionicons name="arrow-back" color="#3F414E" size={35} onPress={onPress} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: DEVICE_WIDTH * 0.15,
    height: DEVICE_WIDTH * 0.15,
    borderRadius: DEVICE_WIDTH * 0.075,
    alignItems: "center",
    justifyContent: "center",
    borderColor: "#EBEAEC",
    borderWidth: 0.7,
  },
});
