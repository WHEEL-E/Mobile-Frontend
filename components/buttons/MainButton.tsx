import React from "react";
import { StyleSheet, View, TouchableOpacity, Text, Image } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { MainButtonProps } from "../../utilities/buttonsUtils";

export const MainButton = (props: MainButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress, icon, image } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <View style={styles.imageContainer}>
        {icon && (
          <Ionicons name={icon.name} size={icon.size} color={icon.color} />
        )}
        {image && <Image source={image.url} style={{ ...styles.image }} />}
      </View>
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    width: "60%",
    height: "60%",
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "center",
  },
});
