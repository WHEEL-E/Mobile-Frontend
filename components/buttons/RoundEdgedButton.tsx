import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { roundEdgedButtonProps } from "../../utilities/types/buttonTypes";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";

const RoundEdgedButton = (props: roundEdgedButtonProps) => {
  const { title, backgroundColor, onPress } = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{ ...styles.button, backgroundColor }}
    >
      <Text style={styles.titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

export default RoundEdgedButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 55,
    width: "80%",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 30,
    backgroundColor: "green",
    alignSelf: "center",
  },
  titleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: DEVICE_WIDTH * 0.05,
  },
});
