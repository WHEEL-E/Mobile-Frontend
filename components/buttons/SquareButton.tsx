import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { SquareButtonProps } from "../../utilities/types/buttonTypes";

export const SquareButton = (props: SquareButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 10,
  },
});
