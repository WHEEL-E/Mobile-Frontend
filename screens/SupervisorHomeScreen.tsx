import React from "react";
import { RootState } from "../store/reducers/rootReducer";
import { StyleSheet, Text, View, ImageBackground } from "react-native";
import { SupervisorHomeProps } from "../utilities/types/navigationTypes/mainNavigationTypes";
import { MainButton } from "../components/buttons/MainButton";
import colors from "../utilities/constants/colors";
import { useTranslation } from "react-i18next";
import { useSelector } from "react-redux";

const SupervisorHomeScreen = (props: SupervisorHomeProps) => {
  const { navigation } = props;
  const { t } = useTranslation();
  const userName = useSelector(
    (state: RootState) => state.user.userData?.mainData.userName
  )?.split(" ")[0];

  return (
    <View style={styles.container}>
      <ImageBackground
        source={require("../assets/images/Header.png")}
        resizeMode="cover"
        style={styles.coverImage}
      />
      <View>
        <Text style={styles.title}>
          {t("supervisorHomeScreen.morning")} {userName}
        </Text>
        <Text style={styles.normalText}>
          {t("supervisorHomeScreen.wishGoodDay")}
        </Text>
      </View>

      <View style={styles.buttons}>
        <View style={styles.buttonColContainer}>
          <MainButton
            title={t("supervisorHomeScreen.profile")}
            buttonStyle={styles.profileButtonStyle}
            titleStyle={styles.profileTitleStyle}
            onPress={() => props.navigation.navigate("Profile")}
            icon={{ name: "ios-person", size: 30, color: colors.darkGreen }}
          />
          <MainButton
            title={t("supervisorHomeScreen.newPatient")}
            buttonStyle={styles.patientButtonStyle}
            titleStyle={styles.patientTitleStyle}
            onPress={() => navigation.navigate("PatientHome")}
            image={{
              url: require("../assets/images/new-patient.png"),
            }}
          />
        </View>
        <View style={styles.buttonColContainer}>
          <MainButton
            title={t("supervisorHomeScreen.patientsStatus")}
            buttonStyle={styles.statusButtonStyle}
            titleStyle={styles.statusTitleStyle}
            onPress={() => navigation.navigate("AssociatedPatients")}
            image={{
              url: require("../assets/images/health-record.png"),
            }}
          />
          <MainButton
            title={t("supervisorHomeScreen.help")}
            buttonStyle={styles.helpButtonStyle}
            titleStyle={styles.helpTitleStyle}
            onPress={() => navigation.navigate("PatientHome")}
            icon={{ name: "ios-help-circle", size: 30, color: "white" }}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 0.9,
    alignItems: "center",
  },
  title: {
    fontSize: 20,
    fontFamily: "Cairo-Bold",
    textAlign: "center",
  },
  normalText: {
    fontSize: 15,
    fontFamily: "Cairo-Regular",
    textAlign: "center",
  },
  buttons: {
    flexDirection: "row",
    width: 200,
    justifyContent: "space-evenly",
    alignItems: "center",
    marginBottom: 15,
  },
  buttonColContainer: {
    flexDirection: "column",
  },
  coverImage: {
    flex: 1,
    width: 380,
    height: 350,
  },
  profileButtonStyle: {
    backgroundColor: "#E7D9EA",
    borderWidth: 2,
    borderColor: "#E7D9EA",
    borderRadius: 30,
    justifyContent: "center",
    padding: 15,
    height: 100,
    width: 150,
    marginHorizontal: 70,
    marginVertical: 5,
  },
  profileTitleStyle: {
    fontFamily: "Cairo-Bold",
    color: colors.darkGreen,
    fontSize: 15,
    textAlign: "center",
  },
  patientButtonStyle: {
    backgroundColor: colors.darkGreen,
    borderWidth: 2,
    borderColor: colors.darkGreen,
    borderRadius: 30,
    justifyContent: "center",
    padding: 15,
    height: 150,
    width: 150,
    marginHorizontal: 70,
    marginVertical: 5,
  },
  patientTitleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: 18,
    textAlign: "center",
  },
  statusButtonStyle: {
    backgroundColor: colors.darkGreen,
    borderWidth: 2,
    borderColor: colors.darkGreen,
    borderRadius: 30,
    justifyContent: "center",
    padding: 15,
    height: 150,
    width: 150,
    marginHorizontal: 70,
    marginVertical: 5,
  },
  statusTitleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: 18,
    textAlign: "center",
  },
  helpButtonStyle: {
    backgroundColor: colors.darkPink,
    borderWidth: 2,
    borderColor: colors.darkPink,
    borderRadius: 30,
    justifyContent: "center",
    padding: 15,
    height: 100,
    width: 150,
    marginHorizontal: 70,
    marginVertical: 5,
  },
  helpTitleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: 20,
    textAlign: "center",
  },
});

export default SupervisorHomeScreen;
