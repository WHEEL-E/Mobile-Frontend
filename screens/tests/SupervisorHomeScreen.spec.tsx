import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import * as redux from "react-redux";
import * as mainButton from "../../components/buttons/MainButton";
import { MainButton } from "../../components/buttons/MainButton";
import SupervisorHomeScreen from "../SupervisorHomeScreen";

jest.mock("react-i18next", () => ({
  ...jest.requireActual("react-i18next"),
  useTranslation: () => {
    return {
      t: () => "translated text",
    };
  },
}));

describe("supervisorHomeScreen", () => {
  beforeAll(() => {
    jest
      .spyOn(redux, "useSelector")
      .mockImplementation(() => "mockFirstName mockLastName");
  });
  it("makes sure screen matches snapShot", () => {
    const dummyProp: any = () => {};
    const supervisorHomeScreen = renderer
      .create(<SupervisorHomeScreen navigation={dummyProp} route={dummyProp} />)
      .toJSON();
    expect(supervisorHomeScreen).toMatchSnapshot();
  });

  it("makes sure button componenets are called", () => {
    jest.spyOn(mainButton, "MainButton");

    const dummyProp: any = () => {};
    render(<SupervisorHomeScreen navigation={dummyProp} route={dummyProp} />);
    expect(MainButton).toBeCalledTimes(4);
  });

  it("makes sure text is rendered properly", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <SupervisorHomeScreen navigation={dummyProp} route={dummyProp} />
    );

    const welcomeText = getByTestId("welcomeText");
    expect(welcomeText.children).toContain("translated text");
  });

  it("makes sure first name is rendered", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <SupervisorHomeScreen navigation={dummyProp} route={dummyProp} />
    );

    const welcomeText = getByTestId("morningText");
    expect(welcomeText.children).toContain("mockFirstName");
    expect(welcomeText.children).not.toContain("mockLastName");
  });
});
