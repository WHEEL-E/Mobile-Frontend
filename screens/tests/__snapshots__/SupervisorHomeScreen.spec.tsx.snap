// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supervisorHomeScreen makes sure screen matches snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.9,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": 350,
        "width": 380,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 350,
            "width": 380,
          },
          undefined,
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Bold",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
      testID="morningText"
    >
      translated text
       
      mockFirstName
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Regular",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
      testID="welcomeText"
    >
      translated text
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginBottom": 15,
        "width": 200,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7D9EA",
            "borderColor": "#E7D9EA",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 100,
            "justifyContent": "center",
            "marginHorizontal": 70,
            "marginVertical": 5,
            "opacity": 1,
            "padding": 15,
            "width": 150,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "color": "#11698E",
              "fontFamily": "Cairo-Bold",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "#11698E",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 70,
            "marginVertical": 5,
            "opacity": 1,
            "padding": 15,
            "width": 150,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "#11698E",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 70,
            "marginVertical": 5,
            "opacity": 1,
            "padding": 15,
            "width": 150,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B91646",
            "borderColor": "#B91646",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 100,
            "justifyContent": "center",
            "marginHorizontal": 70,
            "marginVertical": 5,
            "opacity": 1,
            "padding": 15,
            "width": 150,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
  </View>
</View>
`;
